src/VisibilityDetectingView.tsx

```diff
@@ -41,37 +39,44 @@ export default class VisibilityDetectingView extends PureComponent<Props> {
     if (this.view && UIManager.measure) {
       this.view.measure(this.onMeasure);
     }
-  }
-
-  onMeasure =
-    (x: number, y: number, width: number, height: number, pageX: number, pageY: number) => {
-      // Grab metrics
-      const windowDimensions = Dimensions.get('window');
-      const bottom = pageY + height;
-      const left = pageX;
-      const right = pageX + width;
-      const top = pageY;
-      const windowHeight = windowDimensions.height;
-      const windowWidth = windowDimensions.width;
-
-      // Calculate visibility
-      const visible = (right > 0) && (left < windowWidth) && (top < windowHeight) && bottom > 0;
-
-      // Either trigger onVisible or onInvisible
-      if (!this.previouslyVisible && visible) {
-        if (this.props.onVisible) {
-          this.props.onVisible();
-        }
-      } else if (this.previouslyVisible && !visible) {
-        if (this.props.onInvisible) {
-          this.props.onInvisible();
-        }
+  };
+
+  onMeasure = (
+    x: number,
+    y: number,
+    width: number,
+    height: number,
+    pageX: number,
+    pageY: number,
+  ) => {
```

Comment:
return type?

Reply:
You don't have to specify a return type explicitly, and it's not convention that return types always needs to be specified. 

In this case the compiler can infer that it returns `void`.

Reply:
Sure, but if it's part of a public API, we should err on the side of being explicit.

Reply:
That's true, but I don't think this particular class is part of Hyperview's public API.